dist/smoothscroll.js

```diff
@@ -99,22 +99,37 @@
      */
     function findScrollableParent(el) {
       var isBody;
-      var hasScrollableSpace;
-      var hasVisibleOverflow;
+      var hasVerticalScrollableSpace;
+      var hasHorizontalScrollableSpace;
+      var hasScrollableOverflowY;
+      var hasScrollableOverflowX;
+      var isVerticallyScrollable;
+      var isHorizontallyScrollable;
 
       do {
         el = el.parentNode;
 
         // set condition variables
         isBody = el === d.body;
-        hasScrollableSpace =
-          el.clientHeight < el.scrollHeight ||
-          el.clientWidth < el.scrollWidth;
-        hasVisibleOverflow =
-          w.getComputedStyle(el, null).overflow === 'visible';
-      } while (!isBody && !(hasScrollableSpace && !hasVisibleOverflow));
 
-      isBody = hasScrollableSpace = hasVisibleOverflow = null;
+        hasVerticalScrollableSpace = el.clientHeight < el.scrollHeight;
+        hasHorizontalScrollableSpace = el.clientWidth < el.scrollWidth;
+        hasScrollableOverflowY =
+          w.getComputedStyle(el, null).overflowY === 'auto' ||
```

Comment:
It would be good to create a test case for this, but I got the feeling that the `auto`check could case a false positive, not all `auto` overflow will have scrolling to do and this will prevent the script to scroll body.